### ISSUE TITLE
Docker Implementation for containerization purpose 

### ISSUE THREAD

**Issue by RandomSummer** [Multiverse-of-Projects/NewsAI#74]
Hey! @Devasy23 

I noticed that the repository does not contain a docker and docker-compose.yml file, which is essential for streamlining the setup of the backend services, particularly for the Streamlit projects.

I have extensive experience in setting up Docker environments for local development. I've worked on several projects where docker played a crucial role in streamlining the development process, and I am confident that I can deliver a high-quality solution for this repository.

The addition of a docker-compose.yml file would greatly improve the development and deployment process by enabling easier orchestration of the backend services using Docker.

Proposed Solution
I would like to contribute by adding a Dockerfile and docker-compose.yml file that:

Defines services for the backend.
Configures networking to allow smooth communication.
No one ever needs to worry about the setup it will be done using only one code.
This will allow contributors and developers to quickly spin up the backend environment using Docker, enhancing productivity and minimizing setup errors.

Please let me know if this sounds like a good enhancement, and I'd be happy to get started! 😊

For reference my previous PR: https://github.com/yashasvini121/predictive-calc/issues/45

Thank you for your consideration!

**Comment by RandomSummer** [comment]
Please add the labels gssoc-ext and hacktoberfest label. @Devasy23

**Comment by Devasy23** [comment]
I suggest you first run the project in your local before starting to containerise it

**Comment by RandomSummer** [comment]
sure I will do that

**Comment by Devasy23** [comment]
Return to me one it's done ✅

**Comment by RandomSummer** [comment]
sure I have just created the architecture of my idea, I will send you after I am done.

**Comment by Devasy23** [comment]
Sure 😃

**Comment by Devasy23** [comment]
Any update on the architecture @RandomSummer ? I would love to see your understanding of our architecture.

**Comment by RandomSummer** [comment]
I have analysed the whole project and run it locally. It's a great project, and it will help many people. I have my prototype architecture ready, and I have already started the work. As you have imported the src package in the dashboard(app.py) file, it's a bit tricky for me to handle, but I will find a way to overcome it.

The proposed architecture is:
![Docker Architecture drawio](https://github.com/user-attachments/assets/0ca3665c-9247-4ead-b342-b1e8dac46b8f)

**Comment by Devasy23** [comment]
We'll also need mongodb in this
> I have analysed the whole project and run it locally. It's a great project, and it will help many people. I have my prototype architecture ready, and I have already started the work. As you have imported the src package in the dashboard(app.py) file, it's a bit tricky for me to handle, but I will find a way to overcome it.
> 
> The proposed architecture is:
> ![Docker Architecture drawio](https://github.com/user-attachments/assets/0ca3665c-9247-4ead-b342-b1e8dac46b8f)
>

**Comment by Devasy23** [comment]
@RandomSummer It's been long since you last connected, I hope everything is going well with this issue

**Comment by RandomSummer** [comment]
I have added the dockerfile and docker-compose.yml file, but there has been an error while merging.

**Comment by RandomSummer** [comment]
@Devasy23 Hey bro I requested a merge a week ago but can't merge, 
1. There has been some issue regarding reviews at least one.
2.  There is a greeting error please look after the error. Kindly merge my request.
![newai](https://github.com/user-attachments/assets/7506f03c-1997-49c0-b0cb-58f94bae9f96)

**Comment by Devasy23** [comment]
Is docker file working well, I m planning to deploy it on ec2 free tier

**Comment by RandomSummer** [comment]
yeah, it works well, but I am adding some features like you said "I want to add a local MongoDB via docker and import data into it such that dashboard has data to work with." so wait a little

**Comment by RandomSummer** [comment]
@Devasy23 Hey bro, do you need MongoDB as a separate services, or do you have specific scripts that run in the primary service to connect with MongoDB? Just let me know I will make changes accordingly and I will fix the bugs as well

**Comment by Devasy23** [comment]
We'll need mongodb as seperate service, a sibling service to streamlit service.
And also setup in such a way that if we will pass .env file while running docker compose up -d

**Comment by Devasy23** [comment]
@RandomSummer bro i hope everything is going well, we can connect if you need more clarity regarding any points

**Comment by RandomSummer** [comment]
@Devasy23 I will be doing it within two days

**Comment by RandomSummer** [comment]
Already working on it.

**Comment by Devasy23** [comment]
Okay tell me once you are ready, I'll review it

**Comment by Devasy23** [comment]
@RandomSummer is it ready?

**Comment by Devasy23** [comment]
@RandomSummer is it ready ?

**Comment by Devasy23** [comment]
as today is the last day of gssoc-extd
